oauth2client/client.py

```diff
@@ -598,19 +598,9 @@ def new_request(uri, method='GET', body=None, headers=None,
 
             # Clone and modify the request headers to add the appropriate
             # Authorization header.
-            if headers is None:
-                headers = {}
-            else:
-                headers = dict(headers)
+            headers = self._initialize_headers(headers)
```

Comment:
Good catch.  I doubt this is an issue, but we should avoid potentially breaking changes when possible.